When an IGMP leave message is received, the querier will send a group specific
query message which will cause all snooping switches to reduce the timeout of
that group for all ports to be last membership count, due to which the port
group could age out incorrectly on one of the CLAG switch. Fix: dont age out
port group faster for dually connected hosts or if learnt on peer link

diff --git a/net/bridge/br_multicast.c b/net/bridge/br_multicast.c
index 72b35dc..0996c69 100644
--- a/net/bridge/br_multicast.c
+++ b/net/bridge/br_multicast.c
@@ -1313,8 +1313,9 @@ static int br_ip4_multicast_query(struct net_bridge *br,
 	     pp = &p->next) {
 		if (timer_pending(&p->timer) ?
 		    time_after(p->timer.expires, now + max_delay) :
-		    try_to_del_timer_sync(&p->timer) >= 0)
+		    try_to_del_timer_sync(&p->timer) >= 0) {
 			mod_timer(&p->timer, now + max_delay);
+		}
 	}
 
 out:
